src/libmongoc/src/mongoc/mongoc-client.c

```diff
@@ -3080,3 +3088,33 @@ mongoc_client_enable_auto_encryption (mongoc_client_t *client,
    }
    return _mongoc_cse_client_enable_auto_encryption (client, opts, error);
 }
+
+bool
+mongoc_client_set_timeout_ms (mongoc_client_t *client,
+			      int64_t timeout,
+			      bson_error_t *error)
+{
+   BSON_ASSERT_PARAM (client);
+
+   if (timeout < 0) {
+      bson_set_error (error,
+		      MONGOC_ERROR_TIMEOUT,
+		      MONGOC_ERROR_TIMEOUT_INVALID,
+		      "timeoutMS must be a non-negative integer");
+	 return false;
+   }
+
+   return mongoc_timeout_set_timeout_ms (client->timeout_ms, timeout);
+}
+
+int64_t
+mongoc_client_get_timeout_ms (const mongoc_client_t *client)
+{
+   BSON_ASSERT_PARAM (client);
+
+   if (!mongoc_timeout_is_set (client->timeout_ms)) {
+      return 0;
+   }
```

Comment:
We should also allow the user to distinguish between unset and `0`. The only way I can think of doing that is by exposing `timeout->is_set`.

Reply:
Yes, I suppose they may want to know that as well.  I'll add an additional getter for whether the timeout is set.